prov/sockets/src/sock_cntr.c

```diff
@@ -274,13 +277,13 @@ static int sock_cntr_wait(struct fid_cntr *cntr, uint64_t threshold,
 		goto out;
 	}
 
-	if (_cntr->is_waiting) {
+	if (atomic_get(&_cntr->num_waiting) &&
+			_cntr->domain->progress_mode == FI_PROGRESS_MANUAL) {
```

Comment:
Is `sock_cntr_progress` thread safe? It looks like it is at a glance, but I'm not familiar enough with the surrounding code-base to make that judgement.